### PR TITLE
perf: rewrite opa-executor to bootstrap a built opa instance

### PR DESCRIPTION
**Pull request by suchapalaver**
Rewrites, refactoring in places, Chronicle's `OpaExecutor` and `PolicyBuilder` implementations with a view to bootstrapping a built `Opa` instance.

Adds benchmarks for `OpaExecutor` and `PolicyBuilder` methods. Run `cargo bench` for output.

[CHRON-223](https://blockchaintp.atlassian.net/browse/CHRON-223)

Signed-off-by: Joseph Livesey <joseph.livesey@btp.works>

[CHRON-223]: https://blockchaintp.atlassian.net/browse/CHRON-223?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mtbc** [comment]
When `cfg!(feature = "devmode") && !options.is_present("opa-rule")` might be good to log warn of insecure operating mode.

**Comment by mtbc** [comment]
It may be nice to `#[instrument]` the `build_opa` method given that it might take even rather longer for a non-trivial policy.

**Comment by suchapalaver** [comment]
Thanks for those suggestions, @mtbc!

**Comment by mtbc** [comment]
Thank you for effecting them. :smile:

File: crates/chronicle/src/bootstrap/mod.rs [hunk 1]
```diff
@@ -212,22 +229,15 @@ pub async fn graphql_server<Query, Mutation>(
     options: &ArgMatches,
     jwks_uri: Option<Url>,
     id_pointer: Option<String>,
-    opa_executor: WasmtimeOpaExecutor,
+    opa: Arc<Mutex<WasmtimeOpaExecutor>>,
```
Comment (ryanbtp) [inline crates/chronicle/src/bootstrap/mod.rs:232]: This is fine for now, but we may want to look at moving this to process in spawn_blocking depending on how our profiling turns out

**Comment by ryanbtp** [comment]
For a performance PR, a benchmark is a great thing to add (particularly for something like this that's readily microbenchmarked). Unfortunately bench is still nightly. Have a look at writing a https://github.com/bheisler/criterion.rs test for first / subsequent executions

**Review by ryanbtp — APPROVED** [review]

File: crates/common/src/lib.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#![cfg_attr(feature = "strict", deny(warnings))]
+// #![cfg_attr(feature = "strict", deny(warnings))]
```
Comment (mtbc) [inline crates/common/src/lib.rs:1]: Do we want to leave this in as-is?

Comment (suchapalaver) [reply crates/common/src/lib.rs]: Agh! No! Thanks for catching that!

**Review by mtbc — APPROVED** [review]

**Pull request merged**